src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java

```diff
@@ -0,0 +1,154 @@
+                        )
+                )
+                .expectNext(fieldsDto)
+                .verifyComplete();
+    }
+
+    @Test
+    void shouldRetrieveFieldsDto() {
+        FieldsDto fieldsDto = getFieldsDto();
+
+        UUID correlationUUID = UUID.randomUUID();
+        CreateHostedOrderRequestDto requestDto = getRequestDto();
+
+        Mockito.when(
+                paymentServicesApi.apiOrdersBuildPost(
+                        correlationUUID,
+                        requestDto
+                )
+        ).thenReturn(Mono.just(fieldsDto));
+
+        StepVerifier
+                .create(npgClient.buildOrders(correlationUUID, requestDto))
+                .expectNext(fieldsDto)
+                .verifyComplete();
+    }
+
+    @Test
+    void shouldThrowException() {
+        UUID correlationUUID = UUID.randomUUID();
+        CreateHostedOrderRequestDto requestDto = getRequestDto();
+
+        Mockito.when(
+                paymentServicesApi.apiOrdersBuildPost(
+                        correlationUUID,
+                        requestDto
+                )
+        )
+                .thenReturn(
+                        Mono.error(
+                                new WebClientResponseException(
+                                        "Test error when calling apiOrdersBuildPost",
+                                        HttpStatus.BAD_REQUEST.value(),
+                                        HttpStatus.BAD_REQUEST.getReasonPhrase(),
+                                        null,
+                                        null,
+                                        null
+                                )
+                        )
+                );
+
+        StepVerifier
+                .create(npgClient.buildOrders(correlationUUID, requestDto))
+                .expectError(NpgResponseException.class)
+                .verify();
+    }
+
+    private CreateHostedOrderRequestDto getRequestDto() {
+        return new CreateHostedOrderRequestDto()
+                .version("2")
+                .merchantUrl("localhost/merchant")
+                .order(
+                        new OrderDto()
+                                .orderId("orderId")
+                                .amount("0")
+                                .currency("EUR")
+                                .customerId("customerId")
+                )
+                .paymentSession(
+                        new PaymentSessionDto()
+                                .paymentService("CARDS")
+                                .amount("0")
+                                .actionType(ActionTypeDto.VERIFY)
+                                .language("ITA")
+                                .cancelUrl("localhost/cancel")
+                                .notificationUrl("localhost/notification")
+                                .resultUrl("localhost/result")
+                );
+    }
+
+    private FieldsDto getFieldsDto() {
```

Comment:
Naming:

```suggestion
    private FieldsDto buildTestFieldsDto() {
```

Reply:
Also, you could just make this a constant given that it doesn't depend on the test state